include/z64collision_check.h

```diff
@@ -9,245 +9,391 @@
+#define OC_UNK7 (1 << 7) // Apparently unused
+#define OC_ALL (OC_PLAYER | OC_TYPE1 | OC_TYPE2) // Can have collisions with all three OC types
+
+#define OT_NONE 0 // Has no OC type
+#define OT_HIT_PLAYER (1 << 0) // Had an OC collision with OC type player
+#define OT_UNK1 (1 << 1) // Prevents OC collisions with OT_UNK2. Some horses and toki_sword have it.
+#define OT_UNK2 (1 << 2) // Prevents OC collisions with OT_UNK1. Nothing has it.
+#define OT_PLAYER (1 << 3) // Has OC type player
+#define OT_TYPE1 (1 << 4) // Has OC type 1
+#define OT_TYPE2 (1 << 5) // Has OC type 2
+#define OT_FIRST_ONLY (1 << 6) // Skips AC checks on elements after the first collision. Only used by Ganon
+#define OT_UNK7 (1 << 7) // Apparently unused
+
+#define TOUCH_OFF 0 // Cannot have AT collisions
+#define TOUCH_ON (1 << 0) // Can have AT collisions
+#define TOUCH_HIT (1 << 1) // Had an AT collision
+#define TOUCH_NEAREST (1 << 2) // If a Quad, only collides with the closest bumper
+#define TOUCH_SFX_NORMAL (0 << 3) // Hit sound effect based on AC collider's type
+#define TOUCH_SFX_HARD (1 << 3) // Always uses hard deflection sound
+#define TOUCH_SFX_WOOD (2 << 3) // Always uses wood deflection sound
+#define TOUCH_SFX_NONE (3 << 3) // No hit sound effect
+#define TOUCH_AT_HITMARK (1 << 5) // Draw hitmarks for every AT collision
+#define TOUCH_DREW_HITMARK (1 << 6) // Already drew hitmark for this frame
+#define TOUCH_UNK7 (1 << 7) // Unknown purpose. Used by some enemy quads
+
+#define BUMP_OFF 0 // Cannot have AC collisions
+#define BUMP_ON (1 << 0) // Can have AC collisions
+#define BUMP_HIT (1 << 1) // Had an AC collision
+#define BUMP_HOOKABLE (1 << 2) // Can be hooked if actor has hookability flags set.
+#define BUMP_NO_AT_INFO (1 << 3) // Does not give its info to the AT collider that hit it.
+#define BUMP_NO_DAMAGE (1 << 4) // Does not take damage.
+#define BUMP_NO_SWORD_SFX (1 << 5) // Does not have a sound when hit by player-attached AT colliders.
+#define BUMP_NO_HITMARK (1 << 6) // Skips hit effects.
+#define BUMP_DRAW_HITMARK (1 << 7) // Draw hitmark for AC collision this frame.
+
+#define OCELEM_OFF 0 // Cannot have OC collisions
+#define OCELEM_ON (1 << 0) // Can have OC collisions
+#define OCELEM_HIT (1 << 1) // Had an OC collision
+#define OCELEM_UNK2 (1 << 2) // Apparently unused
+#define OCELEM_UNK3 (1 << 3) // Unknown purpose. Used by Dead Hand element 0 and Dodongo element 5
+#define OCELEM_UNK4 (1 << 4) // Apparently unused
+#define OCELEM_UNK5 (1 << 5) // Apparently unused
+#define OCELEM_UNK6 (1 << 6) // Apparently unused
+#define OCELEM_UNK7 (1 << 7) // Apparently unused
+
+#define OCLINE_NONE 0 // Did not have an OcLine collision
+#define OCLINE_HIT (1 << 0) // Had an OcLine collision
+
+#define DMG_ENTRY(damage, effect) ((damage) | ((effect) << 4))
+
+// These flags are not to be used in code until we figure out how we want to format them. They are only here for reference 
+#define DAMAGE_DEKU_NUT     (1 << 0x00)
```

Comment:
I suggest shortening the prefix to `DMG`, and reordering common terms to be `ARROW_FIRE` instead of `FIRE_ARROW` for instance (like we did with item enums and such).

I also suggest using decimal instead of hex for the shifts, like usual flags, but that's not very important.

Reply:
Would that also be `DMG_KOKIRI_SLASH -> DMG_SLASH_KOKIRI`?

Reply:
I think so yeah